### ISSUE TITLE
数据读取与模型训练

### ISSUE THREAD

**Issue by w18840136818** [ShikangTao/TMACNet#2]
您好！
近期，我在复现 GitHub - TMACNet 项目时遇到了问题。bubbliiiing/yolov8-pytorch项目的数据读取和模型训练代码是针对单图像设计的，而此研究需要处理时序图像输入，需要将不同时间点的图像作为一个输入序列，通过模型检测出目标在时序上的变化情况。然而，目前我还没有找到合适的方法来对数据集进行处理和修改模型以适应这种时序图像输入的需求。如何对数据集进行处理和模型进行训练能否提供更为具体的代码，或者对问题进行一些解答。

**Comment by ShikangTao** [comment]
基于bubbliiiing/yolov8-pytorch项目的数据读取模块（utils/dataloader.py），在annotation_lines读取时相1影像及标签时，新增annotation_lines2对时相2读取，所有后续操作和时相1相同。需要注意的点：
1）使用相同的随机数种子，确保进行mosaic或mixup操作时采样取出的每个批次中影像是一一对应。
        if self.mosaic and self.rand() < self.mosaic_prob and self.epoch_now < self.epoch_length * self.special_aug_ratio:
            randnum = randint(0,100)
            seed(randnum)
            lines = sample(self.annotation_lines, 3)
            lines.append(self.annotation_lines[index])
            # print(lines)
            seed(randnum)
            lines2 = sample(self.annotation_lines2, 3)
            lines2.append(self.annotation_lines2[index])
            # print(lines2)

            seed(randnum)
            shuffle(lines)
            # print(lines)
            seed(randnum)
            shuffle(lines2)
            # print(lines2)

            image, image2, box, box2 = self.get_random_data_with_Mosaic(lines, lines2, self.input_shape)
            
            if self.mixup and self.rand() < self.mixup_prob:
                seed(randnum)
                lines           = sample(self.annotation_lines, 1)
                seed(randnum)
                lines2           = sample(self.annotation_lines2, 1)

                image_2, image2_2, box_2, box2_2  = self.get_random_data(lines[0], lines2[0],self.input_shape, random = self.train)
                image, box, image2, box2      = self.get_random_data_with_MixUp(image, box, image_2, box_2, image2, box2, image2_2, box2_2)
        else:
            image, image2, box, box2      = self.get_random_data(self.annotation_lines[index], self.annotation_lines2[index], self.input_shape, random = self.train)

        image       = np.transpose(preprocess_input(np.array(image, dtype=np.float32)), (2, 0, 1))
        image2       = np.transpose(preprocess_input(np.array(image2, dtype=np.float32)), (2, 0, 1))
        box         = np.array(box, dtype=np.float32)
        box2         = np.array(box2, dtype=np.float32)
2）对时相1和时相2的目标检测标签读取后，利用IoU计算两时相框的分布对比以得到变化检测标签。
